### ISSUE TITLE
Handling SASS color functions in colorDecorators

### ISSUE THREAD

**Issue by talamaska** [Microsoft/vscode#34446]
The integrated color previewer in VS code doesn't understand the variable value and show black. obviously there is a syntax difference from sass rgba and css rgba, would it be possible to change the behavior according to the syntax?

It was already requested to parse the variables and show their colors and I find that colorize does it pretty well.

Like in the other issue, Atom has plugin for that.

- VSCode Version: 1.16.1
- OS Version: Windows 8

Steps to Reproduce:

1.  Create a scss file
2. type inside 
```scss
$input-border: #cccccc; //handled by colorDecorators and colorize
$input-border-focus: darken($input-border, 10%); //not handled by anyone
$input-border-test: rgba($input-border, 0.7); //not handled correctly by the colorDecorators for sass
$input-border-test1: rgba(0, 0, 0, 0.7); // handled by colorDecorators

$test: $input-border; //handled by colorize
$test2: $input-border-focus;  //not handled by anyone
scss```

Reproduces without extensions: Yes

**Comment by ramya-rao-a** [comment]
@aeschli  Trying to verify this one... What's the solution here?

I dont see much difference except that for one of the cases we stopped showing the color
Below on the left is latest from master, in the right is stable 1.16.1

![image](https://user-images.githubusercontent.com/16890566/30939126-4d75290a-a391-11e7-8246-43f71d303a24.png)

**Comment by aeschli** [comment]
The fix is to only show color decorators on color literals. Values with variables, expressions or functions (sass/less) are skipped.
Getting the resolving 100% correct means to rebuild a lot of functionality from SASS and LESS, which is not planned. Extensions can fill in here.

**Comment by ramya-rao-a** [comment]
Ah, got it. Verified!

**Comment by talamaska** [comment]
Yep, hopefully the gap will be filled from extensions. In any case it is better to not show incorrect data. Thanks for looking into this.
